File: frontend/src/types/resource.ts
export type ResourceType = 'leaflet' | 'video' | 'document' | 'website';

export interface Resource {
  id: string;
  type: ResourceType;
  title: string;
  description: string;
  url: string;
  source: string;
  hazard_categories: string[];
  thumbnail_url?: string;
}

export const resourceTypeLabels: Record<ResourceType, string> = {
  leaflet: '리플릿',
  video: '동영상',
  document: '문서',
  website: '웹사이트',
};

export const resourceTypeIcons: Record<ResourceType, string> = {
  leaflet: '📄',
  video: '🎬',
  document: '📋',
  website: '🌐',
};

File: backend/app/models/resource.py
from pydantic import BaseModel
from typing import List, Optional
from enum import Enum


class ResourceType(str, Enum):
    LEAFLET = "leaflet"
    VIDEO = "video"
    DOCUMENT = "document"
    WEBSITE = "website"


class Resource(BaseModel):
    id: str
    type: ResourceType
    title: str
    description: str
    url: str
    source: str
    hazard_categories: List[str]
    thumbnail_url: Optional[str] = None
